src/main/java/core/basesyntax/service/impl/FruitShopCalculationServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitShopStorage;
+import core.basesyntax.model.AbstractTransaction;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.ShopCalculationService;
+import java.util.List;
+
+public class FruitShopCalculationServiceImpl implements ShopCalculationService<Fruit> {
+    private final FruitShopStorage fruitShopStorage;
+
+    public FruitShopCalculationServiceImpl(FruitShopStorage fruitShopStorage) {
+        this.fruitShopStorage = fruitShopStorage;
+    }
+
+    @Override
+    public FruitShopStorage calculate(List<AbstractTransaction<Fruit>> abstractTransactions) {
+        abstractTransactions.forEach(this::processFruitTransaction);
+        return fruitShopStorage;
```

Comment:
The `calculate` method should return a `ShopStorage<Fruit>` instead of `FruitShopStorage` to maintain consistency with the generic type `T` in the `ShopCalculationService` interface. This change will enhance flexibility and align with the interface definition.